### ISSUE TITLE
Respect custom system images

### ISSUE THREAD

**Issue by alena1108** [rancher/rancher#18029]
We should respect custom system images passed via cluster.yml modifications. In case user wants to test custom images, not the ones Rancher officially supports.

@galal-hussein see if we can just not pull system images from types when passed via cluster.yml

https://github.com/rancher/rancher/blob/master/pkg/controllers/management/clusterprovisioner/provisioner.go#L532

**Comment by alena1108** [comment]
What first appeared to be a simple fix, is not obvious to achieve without the bigger changes. We should address it as a part of https://github.com/rancher/rancher/issues/18041 refactor.

For now, in order to use the custom image, it has to be re-tagged as an official one.

**Comment by poelzi** [comment]
I consider this a bug, not an enhancement. In rke, the installer takes the kubernetes_version as a basis and overwrites the images I specify explicitly. In rancher, the system deletes those data from the yaml file without notice. Not only does in behave not like rke, but also deletes data.
We have serious problems with stock kubernetes that renders our cluster inoperable. I developed a patch and tested it on my personal rke cluster, but since our test/production runs rancher testing this on our test systems became just very complicated.

For me, this is absolute unexpected behavior since rancher uses rke, naturally you would expect this functionality to exist in rancher also. Not being able to overwrite specific images removes the possibility to fix severe bugs in any advanced installations.
As a consultant and devops with over 2 decades experience, this is something that makes me mark software as no-go, since it blocks you fixing things when shtf happend.

**Comment by poelzi** [comment]
I tried to use a custom kontainer-driver-metadata repository as suggested with very little positive outcome. You have use very specific nameing like rancher does otherwise you get validation errors. After that you will run into problems that you will not be able to run multiple versions like two 1.15 versions in parallel because the internal data normalization overwrites the different flavors, so you will end up with one version only.
In conclusion, running any sort of customization of system images or running test/development clusters is fundamentally broken.
